dbt/adapters/athena/connections.py

```diff
@@ -141,6 +143,16 @@ def inner():
 class AthenaConnectionManager(SQLConnectionManager):
     TYPE = "athena"
 
+    @classmethod
+    def data_type_code_to_name(cls, type_code: Union[int, str]) -> str:
+        """
+        Get the string representation of the data type from the Athena metadata. Dbt performs a
+        query to retrieve the types of the columns in the SQL query. Then these types are compared
+        to the types in the contract config, simplified because they need to match what is returned
+        by Athena metadata (we are only interested in the broader type, without subtypes nor granularity).
+        """
+        return type_code.split("(")[0].split("<")[0].upper()
```

Comment:
how do we handle the string vs varchar dilemma? 
What I mean here is: do we have some possibility where we set varchar as contract, and then string is returned?

Reply:
This one is pretty hard to unpack but what I understand from pyathena is that PyAthena uses the [GetQueryResult](https://docs.aws.amazon.com/cli/latest/reference/athena/get-query-results.html) operation under the hood the get the columns info as you can see [here](https://github.com/laughingman7743/PyAthena/blob/master/pyathena/result_set.py#LL363C33-L363C33)

Now how de we know how the `Type` value in`GetQueryResult` is filled with ? I am not sure and nothing in the documentation says anything about that.

My practical experience led me to this varchar instead of string. Indeed, what I can try to do is make it ok if a column is set as varchar or string in the contract, it is the same thing.

Reply:
Indeed it's tricky to see if it works. Don't you think that we can add a piece of code where we map string to varchar?
Something like this:

```
if type_code.lower() == 'string:
   return 'VARCHAR'
```

Doing so we force are sure that in case string is returned we have match from the contracts with a varchar.

Reply:
ah thanks, [this](https://github.com/dbt-labs/dbt-core/blob/f1dddaa6e96b743498440925a55e66a7d5cdacfa/core/dbt/include/global_project/macros/adapters/columns.sql#L49) clarify my doubts. 
> The only workaround is to use the ddl_data_type macro and override this macro so that we can make it work.
sound reasonable